File: day:68/lengthOfLongestSubstring.java
import java.util.*;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        // base case
        if(s.length() == 0) {
            return 0;
        }
        int longest = 0;

        for(int i = 0; i < s.length(); i++) {
            int count = 0;
            HashMap<Character, Boolean> hash = new HashMap<Character, Boolean>();
            for(int j = i; j < s.length(); j++ ) {
                if(hash.get(s.charAt(j)) == null) {
                    count += 1;
                    hash.put(s.charAt(j), true);
                    longest = longest > count? longest: count;
                }
                else {
                    break;
                }
            }
        }
        return longest;
    }
}

File: day:68/lengthOfLongestSubstring.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        #base case
        if(len(s) == 0):
            return 0;
        
        longest = 0;
    
        for i in range(len(s)):
            count = 0;
            hash = {};
            for j in range(i, len(s)):
                if s[j] not in hash:
                    count += 1;
                    hash[s[j]] = 1;
                    longest = max(longest, count);
                else:
                    break;
        return longest;
